### PR TITLE
DM-33918: Create code repository for prompt processing

### PR DESCRIPTION
**Pull request by kfindeisen**
This PR adds in the code removed from lsst-dm/dmtn-219#2. The code has been cleaned up to pass `flake8`, but no attempt has been made to further reconcile it with LSST conventions such as namespacing.

### CODE REVIEW

File: src/tester/upload.py [hunk 1]
```diff
@@ -83,9 +81,6 @@ def main():
     )
     storage_client = storage.Client(PROJECT_ID, credentials=credentials)
     bucket = storage_client.bucket("rubin-prompt-proto-main")
-    batch_settings = pubsub_v1.types.BatchSettings(
```
Comment (ktlim) [inline src/tester/upload.py]: We should probably leave this in and add it to the `PublisherClient` constructor below.

File: ups/prompt-prototype.table [hunk 2]
```diff
@@ -4,5 +4,6 @@
 #   the "base" package.
 setupRequired(base)
 setupRequired(sconsUtils)
+setupRequired(daf_butler)    # Used by middleware_interface module.
```
Comment (ktlim) [inline ups/prompt-prototype.table:7]: I don't think this is ever going to be an eups package, so I'm not sure the `ups` directory is needed at all.

**Review by ktlim — APPROVED** [review]
Thanks for cleaning up some of my leftovers.

File: src/tester/upload.py [hunk 3]
```diff
@@ -83,9 +81,6 @@ def main():
...
-        max_messages=INSTRUMENTS[instrument].n_detectors,
-    )
     publisher = pubsub_v1.PublisherClient(credentials=credentials)
```
Comment (kfindeisen) [inline src/tester/upload.py:84]: @ktlim, is this what it should look like? Google's documentation is shockingly bad.
```suggestion
    publisher = pubsub_v1.PublisherClient(credentials=credentials, per_partition_batching_settings=batch_settings)
```

Comment (ktlim) [reply src/tester/upload.py:84]: See https://googleapis.dev/python/pubsub/latest/publisher/index.html#batching
and https://googleapis.dev/python/pubsub/latest/publisher/api/client.html#module-google.cloud.pubsub_v1.publisher.client

I think this is just `batch_settings=batch_settings`.

Comment (kfindeisen) [reply src/tester/upload.py:84]: Oops, wrong namespace: I was looking at https://cloud.google.com/python/docs/reference/pubsublite/latest/google.cloud.pubsublite.cloudpubsub.publisher_client.PublisherClient.

Comment (ktlim) [reply src/tester/upload.py]: https://cloud.google.com/pubsub/docs/choosing-pubsub-or-lite

**Pull request merged**